src/main/scala/jigg/util/JSONUtil.scala

```diff
@@ -90,10 +92,18 @@ object JSONUtil {
         .map(generateXML) } getOrElse List()
     val attrs: Map[String, String] =
       x.filter { case (k, v) => k != ".tag" && k != "text" && k != ".child" }
-         .map { case (k, v) => (k, v.toString) }
+        // Since the `&` character contained in an escaped string, e.g. `&gt;`,
+        // is automatically escaped, the returned nodes is not semantically equal
+        // to the original nodes.
+        // To avoid this issues, all such strings is unescaped before throw into
+        // the XMLUtil.addAttributes method.
+        .map { case (k, v) =>
+        (k, unescapeMap.foldLeft(v.toString){(text, pair) => text.replaceAll(pair._1, pair._2)})}
```

Comment:
This `foldLeft` and the next similar line (106) could be extracted by a function?